addons/website_sale_comparison/models/website_sale_comparison.py

```diff
@@ -1,31 +1,55 @@
 # -*- coding: utf-8 -*-
+
 from collections import OrderedDict
 
-from odoo import fields, models, _
+from odoo import fields, models
 
 
 class ProductAttributeCategory(models.Model):
     _name = "product.attribute.category"
     _description = "Product Attribute Category"
-    _order = 'sequence'
+    _order = 'sequence, id'
+
+    name = fields.Char("Category Name", required=True, translate=True)
+    sequence = fields.Integer("Sequence", default=10, index=True)
 
-    name = fields.Char("Category Name", required=True)
-    sequence = fields.Integer("Sequence", default=10)
+    attribute_ids = fields.One2many('product.attribute', 'category_id', string="Related Attributes", domain="[('category_id', '=', False)]")
```

Comment:
This is a o2m so each attribute can only have one category.

If the domain is not there, it's possible to select an attribute already assigned to another category and it will change its category on save, except the JS framework doesn't handle that case (if the o2m is displayed as many2many_tags on a list view like it is the case here): it will show an incorrect result until the page is reloaded (like the attributes being on 2 different categories at the same time), which is not ideal.

So the domain is to prevent that issue, but it is also an acceptable functional domain: it is more likely an attribute not yet assigned to a category will be selected, and in the rare case where a category has to be changed, the previous category has to be removed manually first.

Note that this only applies to the o2m field from the category list view (which has been added as an extra), when viewing the corresponding m2o on the attribute views (like before) it's always possible to change the category directly.